Sources/gg.datagram.web-requests.sdPlugin/app.js

```diff
@@ -97,9 +97,20 @@ function sendWebSocket(data) {
     ws.onclose = function(evt) { onClose(this, evt); };
     ws.onopen = function() {
         onOpen(this);
+        const start = new Date();
         ws.send(body);
-        ws.close();
-        showOk(data.context);
+        const readyCloseInterval = setInterval(function() {
+            if (ws.bufferedAmount == 0) {
+                ws.close();
+                showOk(data.context);
+                clearInterval(readyCloseInterval);
+            }
+            else if ((new Date().getTime() - start.getTime()) > 3000) {
```

Comment:
`new Date()` isn't really the proper thing to use for measuring time intervals, since it can and does jump to different values when the system clock updates. Can we use an API that gives monotonically increasing times like [`performance.now()`](https://developer.mozilla.org/en-US/docs/Web/API/Performance/now)?

Reply:
Ah cool I've never come across this performance API before. PR updated with this suggestion 👍